zero_bin/rpc/src/main.rs

```diff
@@ -53,6 +53,14 @@ impl Cli {
             } => {
                 let checkpoint_block_number =
                     checkpoint_block_number.unwrap_or((start_block - 1).into());
+                // Ensure checkpoint_block_number is less than start_block
```

Comment:
This check isn't really super meaningful on its own.
Ideally what we would want to have is:

- if a previous block proof was provided as input, we want to _assert_ that it matches the checkpoint of the previous proof
- if no previous block proof was provided, then we want to _assert_ that `checkpoint_block_number == start_block - 1.